yt_dlp/extractor/theplatform.py

```diff
@@ -295,6 +302,16 @@ def _real_extract(self, url):
 
         formats, subtitles = self._extract_theplatform_smil(smil_url, video_id)
 
+        # With some sites, manifest URL must be forced to extract HLS formats
+        if not traverse_obj(formats, lambda _, v: v['format_id'].startswith('hls')):
+            m3u8_url = update_url(url, query='mbr=true&manifest=m3u', fragment=None)
+            ext = urlhandle_detect_ext(self._request_webpage(
+                HEADRequest(m3u8_url), video_id, 'Checking for additional m3u8 formats', fatal=False))
+            if ext == 'm3u8':
+                m3u8_fmts, m3u8_subs = self._extract_m3u8_formats_and_subtitles(m3u8_url, video_id, fatal=False)
```

Comment:
```suggestion
                m3u8_fmts, m3u8_subs = self._extract_m3u8_formats_and_subtitles(
                    m3u8_url, video_id, m3u8_id='hls', fatal=False)
```

Reply:
Applied these fixes, it now works.
I had a regression on NBC with the previous code, currently testing again to see if this changes the issue (haven't looked into it more closely).

Reply:
OK, that fixed the NBC regresssion. Still have a Scripps Network regression that I'm looking into.

Reply:
...aaaand I found the problem. 404 crashed the entire thread. Easy fix: try-except:pass-else.

Reply:
when `fatal=False`, a 404 will only warn, not error
when you run download tests, it treats warnings as errors
the code was fine before, please revert

Reply:
and please share which test was throwing

Reply:
Right, I had put that in there for a reason. Any way to get the tests to accept a 404 at that point? Because the test is now failing even though it shouldn't.

Reply:
Add `expected_warnings` to the test dict, e.g.
```py
'expected_warnings': ['HTTP Error 404: Not Found'],
```

Reply:
Reverted. The test was (or is now again) `ScrippsNetworks` (only has one test).

Reply:
Yeah, sorry about that. Not sure if it's GitHub or something on my end, but the comments aren't quite always appearing properly.